File: 250514/특정 수와 근접한 합/sum-close-to-particular-number.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        // Please write your code here.
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int S = sc.nextInt();
        int[] arr = new int[N];

        for (int i = 0; i < N; i++) {
            arr[i] = sc.nextInt();
        }

        // S와의 차의 절대값이 가장 작은 경우 -> 그 차이를 출력한다.
        int minDiff = Integer.MAX_VALUE;
        for (int i = 0; i < N; i++) {
            for (int j = i+1; j < N; j++) {
                int sum = 0;
                for (int k = 0; k < N; k++) {
                    if (k == i || k == j) continue;
                    sum += arr[k];
                }
                minDiff = Math.min(minDiff, Math.abs(S-sum));
            }
        }
        System.out.print(minDiff);
    }
}


// 

File: 251022/특정 수와 근접한 합/sum-close-to-particular-number.py
N, S = map(int, input().split())
arr = list(map(int, input().split()))

# Please write your code here.

min_diff = S

for i in range(N) :
    for j in range(i+1,N) :

        sum = 0
        for k in range(N) :
            if k == i or k == j :
                continue

            sum += arr[k]

        min_diff = min(min_diff, abs(S-sum))

print(min_diff)